python/rotate_array.py

```diff
@@ -0,0 +1,62 @@
+class Solution:
+    def rotate(self, nums: List[int], k: int) -> None:
+        """
+        Link to problem: https://leetcode.com/problems/rotate-array/
+        - Given an array, rotate the array to the right by k steps,
+          where k is non-negative.
+        - Do not return anything, modify nums in-place instead.
+
+        Time complexity: O(n)
+        Space complexity: O(n)
+
+        """
+        nums_length = len(nums)
+
+        # No need to rotate
+        if nums_length == 1:
+            return
+
+        # No need to rotate more than the length of an array
+        k =  k % nums_length
+
+        temp = nums.copy()
+        for i in range(nums_length):
+            new_i = i + k
+            if new_i >= nums_length:
+                new_i = new_i - nums_length
+            nums[new_i] = temp[i]
+
+
+from fractions import gcd
+class Solution:
+    def rotate(self, nums: List[int], k: int) -> None:
+        """
+        In-place Solution
+
+        Time complexity: O(n)
+        Space complexity: O(1)
```

Comment:
Do you think this has `O(1)` space complexity? (since it is an in-place solution..?)

Reply:
Yes I think so, it seems like you'll always have a constant amount of space allocated for the variables and it's not dependent on the number of elements

Reply:
Great, I was confused with the leetcode result as it somehow used the same amount of memory space...